tests/std/tests/P0896R4_ranges_alg_move/test.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+#include <range_algorithm_support.hpp>
+
+struct int_wrapper {
+    int val = 10;
+    constexpr int_wrapper() = default;
+    constexpr int_wrapper(int x) : val{x} {}
+    constexpr int_wrapper(int_wrapper&& that) : val{std::exchange(that.val, -1)} {}
+    constexpr int_wrapper& operator=(int_wrapper&& that) {
+        val = std::exchange(that.val, -1);
+        return *this;
+    }
+};
+
+constexpr void smoke_test() {
+    using ranges::move, ranges::move_result, ranges::iterator_t;
+    using std::same_as;
+
+    // Validate that move_result aliases in_out_result
+    STATIC_ASSERT(same_as<move_result<int, double>, ranges::in_out_result<int, double>>);
+
+    // Validate dangling story
+    STATIC_ASSERT(same_as<decltype(move(borrowed<false>{}, static_cast<int*>(nullptr))), move_result<ranges::dangling, int*>>);

```

Comment:
Do we enforce limiting to 120 characters in testing files as well?  If so, this line and line 47 exceed 120 characters below.

Reply:
Yes we do.

Reply:
Note that clang-format generally limits lines to 120, but in certain situations it'll get confused and exceed the limit (I recall seeing this with variable templates). If that happens, or if clang-format has been disabled, validate.cpp won't complain. Perhaps we should add iron-clad enforcement, although it would require a slightly larger change in validate.cpp than usual (since this is a line-based property).